### PR TITLE
Feature page for Managed Block Storage

### PR DESCRIPTION
**Pull request by rollandf**
Changes proposed in this pull request:

- Add feature page for storage offloading support

-

-

I confirm that this pull request was submitted according to the [contribution guidelines](https://github.com/oVirt/ovirt-site/blob/master/CONTRIBUTING.md): @rollandf

This pull request needs review by: @yanivlavi @nirs @bennyz @tnisan

### CODE REVIEW

File: source/develop/release-management/features/storage/cinderlib-integration.html.md [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+feature_status: WIP 4.3
+---
+
+# Storage Offloading
+
+## Summary
+
+Many storage vendors provide a offloading API allowing to do fast storage side actions with minimal network usage from the virt management side. Such APIs already are integrated in Cinder.
+
+This feature enable the user to be able to consume any storage backend supported in Cinder in order to create virtual disks for its VMs, without the need of a full Openstack deployment.
```
Comment (bennyz) [inline source/develop/release-management/features/storage/cinderlib-integration.html.md]: enables

File: source/develop/release-management/features/storage/cinderlib-integration.html.md [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+
+Cinderlib will be packaged in RPM.
+The Cinder drivers and Cinderlib needs to be installed in the Engine.
+OS-Brick needs to installed on the hosts
+
+
+### Open Issues / Limitations
+
+- Cinderlib needs to be packaged in RPM that will be required by the Engine.
+- os-brick needs to be installed on the hosts.
```
Comment (sandrobonazzola) [inline source/develop/release-management/features/storage/cinderlib-integration.html.md]: maybe worth to mention that right now for installing cinderlib you'll need to install the whole openstack-cinder service?

File: source/develop/release-management/features/storage/cinderlib-integration.html.md [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+
+Many storage vendors provide a offloading API allowing to do fast storage side actions with minimal network usage from the virt management side. Such APIs already are integrated in Cinder.
+
+This feature enables the user to be able to consume any storage backend supported in Cinder in order to create virtual disks for its VMs, without the need of a full Openstack deployment.
+
+
+### Owner
+
+*   Fred Rolland (<frolland@redhat.com>)
+*   Benny Zlotnik (<bzlotnik@redhat.com>)
```
Comment (shenitzky) [inline source/develop/release-management/features/storage/cinderlib-integration.html.md]: Can you pleas add me as an owner?

File: source/develop/release-management/features/storage/cinderlib-integration.html.md [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+The Vdsm uses the Cinder's 'os_brick' library to collect the data.
+
+#### Run VM
+
+The user runs a VM with a Cinderlib disk :
+  - The Engine calls a Cinderlib API to connect the volume using the connection information for the specific host and return the needed data for the attachment.
+  - The Engine calls a Vdsm verb to attach the volume with the data needed. The Vdsm stores locally the attachment details that will be needed for detach, and then return the path.
+  - The Engine uses the returned path for the volume in the VM XML.
+
+#### Run VM
```
Comment (shenitzky) [inline source/develop/release-management/features/storage/cinderlib-integration.html.md]: same paragraph as in line #86

File: source/develop/release-management/features/storage/cinderlib-integration.html.md (hunk 2 above)
Comment (rollandf) [reply source/develop/release-management/features/storage/cinderlib-integration.html.md]: Done

File: source/develop/release-management/features/storage/cinderlib-integration.html.md (hunk 3 above)
Comment (rollandf) [reply source/develop/release-management/features/storage/cinderlib-integration.html.md]: Done

File: source/develop/release-management/features/storage/cinderlib-integration.html.md [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+  - The Engine calls a Cinderlib API to connect the volume using the connection information for the specific host and return the needed data for the attachment.
+  - The Engine calls a Vdsm verb to attach the volume with the data needed. The Vdsm stores locally the attachment details that will be needed for detach, and then return the path.
+  - The Engine uses the returned path for the volume in the VM XML.
+
+
+#### Stop VM
+
+The user stops a VM with a Cinderlib disk :
+  - The Engine calls a Vdsm verb to detach the volume with the data needed. The Vdsm use the attachment details stored locally.
+  - The Engine calls a Cinderlib API to disconnect the volume.
```
[the remark this answers is not in the document]
Comment (rollandf) [reply source/develop/release-management/features/storage/cinderlib-integration.html.md]: Shared disks are currently not supported.

File: source/develop/release-management/features/storage/cinderlib-integration.html.md [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+- Add connection information as part of GetCapabilities verb using 'os_brick'.
+- Add an 'attach' verb that will get the details of the volume, attach to the local host, persist the data for attachment and return the path.
+
+// TODO describe additional changes needed
+
+### Engine
+
+- Add DB support for additional Storage Domain information.
+- Block not supported storage operations.
+- Python executable giving API to Cinderlib
```
[the remark this answers is not in the document]
Comment (rollandf) [reply source/develop/release-management/features/storage/cinderlib-integration.html.md]: I am not sure, I will follow up with Gorka

File: source/develop/release-management/features/storage/cinderlib-integration.html.md [hunk 7]
```diff
@@ -0,0 +1,141 @@
...
+The Cinder drivers and Cinderlib needs to be installed in the Engine.
+OS-Brick needs to installed on the hosts
+
+
+### Open Issues / Limitations
+
+- Cinderlib needs to be packaged in RPM that will be required by the Engine.
+- os-brick needs to be installed on the hosts.
+- Currently, for installing Cinderlib it will be necessary to install the whole openstack-cinder libraries.
+- The Engine in oVirt will need to have access to the Storage Management API network.
```
[the remark this answers is not in the document]
Comment (rollandf) [reply source/develop/release-management/features/storage/cinderlib-integration.html.md]: Yes

File: source/develop/release-management/features/storage/cinderlib-integration.html.md [hunk 8]
```diff
@@ -0,0 +1,141 @@
...
+- Not all storage flows are supported in the first release of these feature. For example :
+    - Storage migration (cold & live)
+    - OVF
+    - Master domain
+    - Leases
+    - Storage Domain monitoring
+- Some drivers requires additional packages that will need to be installed on the hosts. See: https://github.com/openstack/cinder/blob/master/driver-requirements.txt
```
[the remark this answers is not in the document]
Comment (rollandf) [reply source/develop/release-management/features/storage/cinderlib-integration.html.md]: No, some of the packages are not open source or not available in the regular channels.
It is the same behavior as in Openstack

File: source/develop/release-management/features/storage/cinderlib-integration.html.md [hunk 9]
```diff
@@ -0,0 +1,144 @@
...
+#### Create Disk
+
+The user creates a disk on the storage domain:
+  - The Engine calls a Cinderlib API with the driver backend parameters, the volume id, the volume size and the DB URL.
+  - If the call to Cinderlib API is successful, the Engine creates the Disk in the DB.
+
+#### Delete Disk
+
+The user delete a disk on the storage domain:
+  - The Engine calls a Cinderlib API with the driver backend parameters, the volume id, the volume size and the DB URL to delete the volume.
```
Comment (shenitzky) [inline source/develop/release-management/features/storage/cinderlib-integration.html.md]: "... the volume size" - not needed

File: source/develop/release-management/features/storage/cinderlib-integration.html.md [hunk 10]
```diff
@@ -0,0 +1,144 @@
...
+  - If the call to Cinderlib API is successful, the Engine deletes the disk in the DB
+
```
Comment (shenitzky) [inline source/develop/release-management/features/storage/cinderlib-integration.html.md]: we can add also the extend disk  operation

File: source/develop/release-management/features/storage/cinderlib-integration.html.md (hunk 9 above)
Comment (rollandf) [reply source/develop/release-management/features/storage/cinderlib-integration.html.md]: Fixed

File: source/develop/release-management/features/storage/cinderlib-integration.html.md (hunk 10 above)
Comment (rollandf) [reply source/develop/release-management/features/storage/cinderlib-integration.html.md]: Done

**Pull request merged**